splight_lib/models/alert.py

```diff
@@ -1,48 +1,207 @@
-from enum import Enum
-from typing import Dict, List, Optional, Union
+import json
+import re
+from enum import Enum, auto
+from typing import List, Literal, Optional
 
-from pydantic import BaseModel, EmailStr, Field
+from pydantic import (
+    BaseModel,
+    Field,
+    PositiveInt,
+    ValidationError,
+    model_validator,
+)
+from strenum import LowercaseStrEnum, UppercaseStrEnum
 from typing_extensions import TypedDict
 
 from splight_lib.constants import DESCRIPTION_MAX_LENGTH
 from splight_lib.models.base import SplightDatabaseBaseModel
+from splight_lib.models.exceptions import (
+    InvalidAlertConfiguration,
+    MissingAlertItemExpression,
+)
 
 
-class AlertStatus(str, Enum):
-    ALERT = "alert"
-    NO_ALERT = "no_alert"
-    NO_DATA = "no_data"
-    DISABLED = "disabled"
+class AlertItemType(UppercaseStrEnum):
+    EXPRESSION = auto()
+    QUERY = auto()
 
 
-class AlertOperator(str, Enum):
+class QueryFilter(TypedDict):
+    id: str
+    name: str
+
+
+class GroupUnit(LowercaseStrEnum):
+    EMPTY = ""
+    MINUTE = auto()
+    HOUR = auto()
+    DAY = auto()
+
+
+class GroupCriteria(LowercaseStrEnum):
+    EMPTY = ""
+    AVG = auto()
+    SUM = auto()
+    MIN = auto()
+    MAX = auto()
+    LAST = auto()
+
+
+class AlertItem(BaseModel):
+    id: Optional[str] = Field(None, max_length=100)
+
+    ref_id: str = Field(max_length=5)
+    type: AlertItemType.QUERY
+
+    expression: str = ""
+    expression_plain: str = ""
+
+    query_filter_asset: Optional[QueryFilter] = None
+    query_filter_attribute: Optional[QueryFilter] = None
+
+    query_group_function: GroupCriteria = GroupCriteria.EMPTY
+    query_group_unit: GroupUnit = GroupUnit.EMPTY
+
+    query_sort_field: str = ""
+    query_sort_direction: int = Field(-1, ge=-1, le=1)
+
+    query_plain: str = ""
+    query_limit: int = Field(10000, ge=1, le=1000)
```

Comment:
The values are inconsistent
```suggestion
    query_limit: int = Field(10000, ge=1, le=10000)
```

Reply:
Check the values with API before merge

Reply:
done